### ISSUE TITLE
Can't process emails - login doesn't have a default value

### ISSUE THREAD

**Issue by pkulas** [zammad/zammad#1020]
Hi, we cannot process lot of mails, sometimes after it doesn't process an email it stops from work and doesn't try to load new mails(need restart of zammad to process new mails). We recently upgraded to 1.5 version, waiting for any changes, will it retry unprocessed mails to check once again? Any way to force check?

### Infos:

* Used Zammad version: 1.4
* Used Zammad installation source: package(deb)
* Operating system: Ubuntu 16.04
* Browser + version: ---

 #<ActiveRecord::StatementInvalid: Mysql2::Error: Field 'login' doesn't have a default value: INSERT INTO `users` VALUES ()> 
[production.log.txt](https://github.com/zammad/zammad/files/964619/production.log.txt)

### Expected behavior:

* Process mails

### Actual behavior:

* Doesn't process lot of mails

### Steps to reproduce the behavior:

* Setup mail settings, wait to fetch list.

**Comment by martini** [comment]
Hi @pkulas 

thanks for reporting this issue. How many emails do you have under /opt/zammad/tmp/unprocessable_mail/ ?

Can you attache 2-3 examples here? 

Per default the email address is used to login. Did your setup already work (or is it a fresh installation)?

 -Martin

**Comment by pkulas** [comment]
>  How many emails do you have under /opt/zammad/tmp/unprocessable_mail/ ?
We have 67 mails under /opt/zammad/tmp/unprocessable_mail/
> Can you attache 2-3 examples here?
[eml-files.zip](https://github.com/zammad/zammad/files/964768/eml-files.zip)
a00a2df77a3e3b94163abc04863941e2.eml is most simple.

>  Did your setup already work (or is it a fresh installation)? 
It's already working by month or more but recently( like a week) we have these issues.

**Comment by martini** [comment]
Thanks. Emails are simple (there should be no issue). Did you change some settings last time? Do you have triggers and/or postmaster filters now?

**Comment by raphaelm** [comment]
Same problem here, 17 mails already, same problem according to log.

Can I help debugging in any way? Is there a way to re-insert the emails into zammad later? Some of them are important to us.

**Comment by raphaelm** [comment]
Full log excerpt:

```
I, [2017-05-10T22:26:31.663576 #31319]  INFO -- : fetching imap (mail.rami.io/info@opacapp.de port=993,ssl=true,folder=)
I, [2017-05-10T22:26:31.896678 #31319]  INFO -- :  - message 1/1
I, [2017-05-10T22:26:32.732390 #31319]  INFO -- : Can't fetch image for '***@gmail.com' (maybe no avatar available), http code: 422
I, [2017-05-10T22:26:32.754131 #31319]  INFO -- : NOTICE create - self.updated_by_id is different: 8903/1
I, [2017-05-10T22:26:32.754508 #31319]  INFO -- : NOTICE create - self.created_by_id is different: 8903/1
E, [2017-05-10T22:26:32.763623 #31319] ERROR -- : can't run postmaster pre filter Channel::Filter::IdentifySender
E, [2017-05-10T22:26:32.763708 #31319] ERROR -- : #<ActiveRecord::StatementInvalid: Mysql2::Error: Field 'login' doesn't have a default value: INSERT INTO `users` VALUES ()>
E, [2017-05-10T22:26:32.767511 #31319] ERROR -- : ERROR: Can't process email, you will find it for bug reporting under /opt/zammad/tmp/unprocessable_mail//0f8afb523e2488d1da325917bea3ce3b.eml, please create an issue at https://github.com/zammad/zammad/issues
```

**Comment by thorsteneckel** [comment]
Hi @raphaelm - can you please provide the raw file `/opt/zammad/tmp/unprocessable_mail//0f8afb523e2488d1da325917bea3ce3b.eml` anonymized here or to support@zammad.com?

In the most cases this is not a general but a specific issue with the format of the mail sender client. We'll have a look. Thanks!

**Comment by raphaelm** [comment]
Hi, I sent two samples to support@zammad.com, thanks!

**Comment by pkulas** [comment]
After upgrading to 1.5 seems it works better, only one mail got unprocessed so far (not any important so that's fine). Unfortunately can't check what was the error message due to Zammad not storing error logs after update, just INFO logs.

**Comment by martini** [comment]
After some investigation, I found out that the error message "can't run postmaster pre filter Channel::Filter::IdentifySender", "Field 'login' doesn't have a default" is issue https://github.com/zammad/zammad/issues/811 (which is fixed now).

@raphaelm Thanks for your examples. The only strange thing of the examples is, that they have all email addresses (so they are note related to this or issue https://github.com/zammad/zammad/issues/811). And I was able to process all this emails without problems. So I guess there is something else. So let's watch if there are still issues.
